src/gt4py/next/otf/workflow.py

```diff
@@ -60,38 +63,105 @@ def __call__(self, inp: StartT_contra) -> EndT_co:
         ...
 
 
-@dataclasses.dataclass(frozen=True)
-class Step(Generic[StartT, EndT]):
+class WorkflowWithReplace(Workflow[StartT, EndT], Generic[StartT, EndT]):
+    """
+    Subworkflow replacement mixin.
+
+    Any subclass MUST be a dataclass for `.replace` to work
     """
-    Workflow step convenience wrapper.
 
-    Can wrap any callable which implements the workflow step protocol,
-    adds the .chain(other_step) method for convenience.
+    def replace(
+        self: WorkflowWithReplace[StartT, EndT], **kwargs: Any
+    ) -> WorkflowWithReplace[StartT, EndT]:
+        """
+        Build a new instance with replaced substeps.
+
+        Raises:
+            TypeError: If `self` is not a dataclass.
+        """
+        if not dataclasses.is_dataclass(self.__class__):
+            raise TypeError(f"{self.__class__} is not a dataclass")
+        return dataclasses.replace(self, **kwargs)
+
+
+class Chainable(Workflow[StartT, EndT], Generic[StartT, EndT]):
```

Comment:
The naming scheme does not match the one used for `Chainable`. I would actually prefer something like
`ReplacableWorkflowMixin`, but your choice.

Reply:
For me personally `ChainableXYZ` and `ReplaceEnabledXYZ` is close enough. Grammatically "chainable" and "replace enabled" are both adjective phrases.

Reply:
I saw that, removed my initial comment from `WorkflowWithReplace` reformulated the text, but forgot to change `ReplacableWorkflowMixin`.... Whatever you like. Also `ChainableWorkflow` and `WorkflowWithReplace` (very close to your initial version) would (have) be(en) fine. I was just after the Workflow appears in one name, but not the other.